cdap-ui/app/cdap/components/PluginJSONCreator/Create/Content/WidgetCollection/WidgetAttributesCollection/WidgetAttributeInput/AttributeMultipleValuesInput/index.tsx

```diff
@@ -0,0 +1,235 @@
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import { withStyles } from '@material-ui/core';
+import { StyleRules } from '@material-ui/core/styles';
+import If from 'components/If';
+import KeyValuePairs from 'components/KeyValuePairs';
+import { COMMON_DELIMITER } from 'components/PluginJSONCreator/constants';
+import PluginInput from 'components/PluginJSONCreator/Create/Content/PluginInput';
+import * as React from 'react';
+
+export enum SupportedType {
+  String = 'string',
+  Number = 'number',
+  ValueLabelPair = 'Value-Label pair',
+  IDLabelPair = 'ID-Label pair',
+}
+
+const styles = (): StyleRules => {
+  return {
+    typeSelectInput: {
+      marginTop: '5px',
+      marginBottom: '5px',
+      width: '20%',
+    },
+  };
+};
+
+export const processSupportedTypes = (types: string[]) => {
+  const allTypes = new Set();
+  types.forEach((type) => {
+    switch (type) {
+      case 'string[]':
+        allTypes.add(SupportedType.String);
+        break;
+      case 'number[]':
+        allTypes.add(SupportedType.Number);
+        break;
+      case 'IOption[]':
+        allTypes.add(SupportedType.IDLabelPair);
+        break;
+      case 'IDropdownOption[]':
+        allTypes.add(SupportedType.String);
+        allTypes.add(SupportedType.Number);
+        allTypes.add(SupportedType.ValueLabelPair);
+        break;
+      case 'ISelectOptions[]':
+        allTypes.add(SupportedType.ValueLabelPair);
+        break;
+      case 'FilterOption[]':
+        allTypes.add(SupportedType.String);
+        allTypes.add(SupportedType.IDLabelPair);
+        break;
+      default:
+        break;
+    }
+  });
+  return allTypes;
+};
+
+const AttributeMultipleValuesInputView = ({
+  classes,
+  supportedTypes,
+  widgetToAttributes,
+  setWidgetToAttributes,
+  widgetID,
+  field,
+}) => {
+  const cleanSupportedTypes = Array.from(processSupportedTypes(supportedTypes).values());
+  const [selectedType, setSelectedType] = React.useState(null);
+  const [currentInput, setCurrentInput] = React.useState(null);
+
+  React.useEffect(() => {
+    const initialType = getInitialType();
+    setSelectedType(initialType);
+    setCurrentInput(getUpdatedCurrentInput(initialType));
+  }, []);
+
+  function getInitialType() {
+    const widgetAttributes = widgetToAttributes[widgetID][field];
+    let newType;
+    if (!widgetAttributes || widgetAttributes.length === 0) {
+      newType = cleanSupportedTypes[0];
+    } else if (widgetAttributes[0].value) {
+      newType = SupportedType.ValueLabelPair;
+    } else if (widgetAttributes[0].id) {
+      newType = SupportedType.IDLabelPair;
+    } else {
+      newType = SupportedType.String;
+    }
+    return newType;
+  }
+
+  function getUpdatedCurrentInput(newType) {
+    const widgetAttributes = widgetToAttributes[widgetID][field];
+    if (!newType) {
```

Comment:
Can we break this `if..else` condition blocks to single level? Too many levels makes it hard to read and grasp the logic right away.

Reply:
I broke them down to functions and added some comments.